Commit 557829e breaks writing image

Broken from 557829ed1fe556a39d048a0e652fbbed1cecc6b5 and beyond, while I did say it worked fine, more commits were added afterwards so the problem's most likely somewhere in there.
```
~/Desktop » waifu2x-converter-cpp --log-level 1 --input 78156819_p0.jpg                    
CUDA: GeForce GTX 1080
Processing file [1/1] "/home/katt/Desktop/78156819_p0.jpg": OpenCV(4.1.2) /build/opencv/src/opencv-4.1.2/modules/imgcodecs/src/loadsave.cpp:661: error: (-2:Unspecified error) could not find a writer for the specified extension in function 'imwrite_'

Done, took: 0s total, file: 1.160s avg: 1.832s
Finished processing 1 files, took: 1.832secs total, filter: 0.937secs; 0 files skipped, 1 files errored. [GFLOPS: 1173.63, GFLOPS-Filter: 2293.80]
```

Compiled with:
GNU 9.2.0, OpenCV 4.1.2, OpenCL 2.2

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -240,15 +240,13 @@ bool validate_format_extension(std::string ext)
 #if defined(_WIN32) && defined(_UNICODE)
 bool validate_format_extension(std::wstring ext_w)
 {
-	std::string ext;
-	
 	if(ext_w.length() == 0)
 		return false;
 	
 	if(ext_w.at(0) == L'.')
 		ext_w=ext_w.substr(1);
 	
-	wstr2str(&ext, &ext_w);
+	std::string ext = wstr2str(ext_w);
 	return validate_format_extension(ext);
 }
 #endif
@@ -896,8 +894,7 @@ int main(int argc, char** argv)
 		cmdPngCompression.getValue()
 	};
 
-	_tstring outputFormat;
-	_str2tstr(&outputFormat, &cmdOutputFormat.getValue());
+	_tstring outputFormat = _str2tstr(cmdOutputFormat.getValue());
 	
 	int convMode = CONV_NONE;
 	

--- a/src/modelHandler.cpp
+++ b/src/modelHandler.cpp
@@ -1188,8 +1188,7 @@ namespace w2xc
 			jsonFile.open(fileName);
 			if (!jsonFile.is_open())
 			{
-				std::string fname;
-				_tstr2str(&fname, &fileName);
+				std::string fname = _tstr2str(fileName);
 				std::cerr << "Error : couldn't open " << fname << std::endl;
 				return false;
 			}

--- a/src/tstring.cpp
+++ b/src/tstring.cpp
@@ -24,20 +24,22 @@
 
 #include "tstring.hpp"
 
-void wstr2str(std::string* s, const std::wstring* ws){
+std::string wstr2str(std::wstring ws){
 	std::setlocale(LC_ALL, "en_US.utf8");
-	char *buf = new char[ws->size()];
-	size_t num_chars = wcstombs(buf, ws->c_str(), ws->size());
+	char *buf = new char[ws.size()];
+	size_t num_chars = wcstombs(buf, ws.c_str(), ws.size());
 	
-	*s = std::string( buf, num_chars );
+	std::string s( buf, num_chars );
 	delete[] buf;
+	return s;
 }
 
-void str2wstr(std::wstring* ws, const std::string* s){
+std::wstring str2wstr(std::string s){
 	std::setlocale(LC_ALL, "en_US.utf8");
-	wchar_t *buf = new wchar_t[s->size()];
-	size_t num_chars = mbstowcs(buf, s->c_str(), s->size());
+	wchar_t *buf = new wchar_t[s.size()];
+	size_t num_chars = mbstowcs(buf, s.c_str(), s.size());
 	
-	*ws = std::wstring( buf, num_chars );
+	std::wstring ws( buf, num_chars );
 	delete[] buf;
+	return ws;
 }

--- a/src/tstring.hpp
+++ b/src/tstring.hpp
@@ -34,17 +34,17 @@ void str2wstr(std::wstring* ws, const std::string* s);
 #if defined(_WIN32) && defined(_UNICODE)
 	typedef	std::wstring		_tstring;
 	typedef	std::wstringstream	_tstringstream;
-	#define _tstr2wstr(T,X) wcscpy(T,X);
+	#define _tstr2wstr(T,X) X;
 	#define _tstr2str(T,X) wstr2str(T,X);
-	#define _wstr2tstr(T,X) wcscpy(T,X);
+	#define _wstr2tstr(T,X) X;
 	#define _str2tstr(T,X) str2wstr(T,X);
 #else
 	typedef	std::string			_tstring;
 	typedef	std::stringstream	_tstringstream;
 	#define _tstr2wstr(T,X) str2wstr(T,X);
-	#define _tstr2str(T,X) strcpy(T,X);
+	#define _tstr2str(T,X) X;
 	#define _wstr2tstr(T,X) wstr2str(T,X);
-	#define _str2tstr(T,X) strcpy(T,X);
+	#define _str2tstr(T,X) X;
 #endif
 
 #if defined(_WIN32) && defined(_UNICODE)

--- a/src/tstring.hpp
+++ b/src/tstring.hpp
@@ -34,17 +34,17 @@ void str2wstr(std::wstring* ws, const std::string* s);
 #if defined(_WIN32) && defined(_UNICODE)
 	typedef	std::wstring		_tstring;
 	typedef	std::wstringstream	_tstringstream;
-	#define _tstr2wstr(T,X) T=X;
+	#define _tstr2wstr(T,X) wcscpy(T,X);
 	#define _tstr2str(T,X) wstr2str(T,X);
-	#define _wstr2tstr(T,X) T=X;
+	#define _wstr2tstr(T,X) wcscpy(T,X);
 	#define _str2tstr(T,X) str2wstr(T,X);
 #else
 	typedef	std::string			_tstring;
 	typedef	std::stringstream	_tstringstream;
 	#define _tstr2wstr(T,X) str2wstr(T,X);
-	#define _tstr2str(T,X) T=X;
+	#define _tstr2str(T,X) strcpy(T,X);
 	#define _wstr2tstr(T,X) wstr2str(T,X);
-	#define _str2tstr(T,X) T=X;
+	#define _str2tstr(T,X) strcpy(T,X);
 #endif
 
 #if defined(_WIN32) && defined(_UNICODE)

--- a/src/tstring.hpp
+++ b/src/tstring.hpp
@@ -34,17 +34,17 @@ void str2wstr(std::wstring* ws, const std::string* s);
 #if defined(_WIN32) && defined(_UNICODE)
 	typedef	std::wstring		_tstring;
 	typedef	std::wstringstream	_tstringstream;
-	#define _tstr2wstr(T,X) X;
+	#define _tstr2wstr(T,X) wcscpy(T,X);
 	#define _tstr2str(T,X) wstr2str(T,X);
-	#define _wstr2tstr(T,X) X;
+	#define _wstr2tstr(T,X) wcscpy(T,X);
 	#define _str2tstr(T,X) str2wstr(T,X);
 #else
 	typedef	std::string			_tstring;
 	typedef	std::stringstream	_tstringstream;
 	#define _tstr2wstr(T,X) str2wstr(T,X);
-	#define _tstr2str(T,X) X;
+	#define _tstr2str(T,X) strcpy(T,X);
 	#define _wstr2tstr(T,X) wstr2str(T,X);
-	#define _str2tstr(T,X) X;
+	#define _str2tstr(T,X) strcpy(T,X);
 #endif
 
 #if defined(_WIN32) && defined(_UNICODE)

--- a/src/tstring.hpp
+++ b/src/tstring.hpp
@@ -34,17 +34,17 @@ void str2wstr(std::wstring* ws, const std::string* s);
 #if defined(_WIN32) && defined(_UNICODE)
 	typedef	std::wstring		_tstring;
 	typedef	std::wstringstream	_tstringstream;
-	#define _tstr2wstr(T,X) wcscpy(T,X);
+	#define _tstr2wstr(T,X) T=X;
 	#define _tstr2str(T,X) wstr2str(T,X);
-	#define _wstr2tstr(T,X) wcscpy(T,X);
+	#define _wstr2tstr(T,X) T=X;
 	#define _str2tstr(T,X) str2wstr(T,X);
 #else
 	typedef	std::string			_tstring;
 	typedef	std::stringstream	_tstringstream;
 	#define _tstr2wstr(T,X) str2wstr(T,X);
-	#define _tstr2str(T,X) strcpy(T,X);
+	#define _tstr2str(T,X) T=X;
 	#define _wstr2tstr(T,X) wstr2str(T,X);
-	#define _str2tstr(T,X) strcpy(T,X);
+	#define _str2tstr(T,X) T=X;
 #endif
 
 #if defined(_WIN32) && defined(_UNICODE)

--- a/src/tstring.hpp
+++ b/src/tstring.hpp
@@ -28,23 +28,23 @@
 #include <clocale>
 #include "tchar.h"
 
-void wstr2str(std::string* s, const std::wstring* ws);
-void str2wstr(std::wstring* ws, const std::string* s);
+std::string wstr2str(std::wstring ws);
+std::wstring str2wstr(std::string s);
 
 #if defined(_WIN32) && defined(_UNICODE)
 	typedef	std::wstring		_tstring;
 	typedef	std::wstringstream	_tstringstream;
-	#define _tstr2wstr(T,X) X;
-	#define _tstr2str(T,X) wstr2str(T,X);
-	#define _wstr2tstr(T,X) X;
-	#define _str2tstr(T,X) str2wstr(T,X);
+	#define _tstr2wstr(X) X;
+	#define _tstr2str(X) wstr2str(X);
+	#define _wstr2tstr(X) X;
+	#define _str2tstr(X) str2wstr(X);
 #else
 	typedef	std::string			_tstring;
 	typedef	std::stringstream	_tstringstream;
-	#define _tstr2wstr(T,X) str2wstr(T,X);
-	#define _tstr2str(T,X) X;
-	#define _wstr2tstr(T,X) wstr2str(T,X);
-	#define _str2tstr(T,X) X;
+	#define _tstr2wstr(X) str2wstr(X);
+	#define _tstr2str(X) X;
+	#define _wstr2tstr(X) wstr2str(X);
+	#define _str2tstr(X) X;
 #endif
 
 #if defined(_WIN32) && defined(_UNICODE)

--- a/src/w2xconv.cpp
+++ b/src/w2xconv.cpp
@@ -588,8 +588,7 @@ void w2xconv_free(void *p)
 
 static void setPathError(W2XConv *conv, enum W2XConvErrorCode code, _tstring const &path)
 {
-	std::string strpath;
-	_tstr2str(&strpath, &path)
+	std::string strpath = _tstr2str(path);
 	clearError(conv);
 
 	conv->last_error.code = code;
@@ -2102,8 +2101,7 @@ void w2xconv_convert_mat
 		std::vector<uchar> imageBuffer;
 		std::wstring ext_w = std::wstring(filepath);
 		ext_w = ext_w.substr(ext_w.find_last_of(L'.'));
-		std::string ext;
-		wstr2str(&ext, &ext_w);
+		std::string ext=wstr2str(ext_w);
 		
 		if(!cv::imencode(ext.c_str(),img, imageBuffer, imwrite_params))
 		{

